# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1316572933 25200
#      Tue Sep 20 19:42:13 2011 -0700
# Node ID cb2b57abcc3541cda0b80223591d6c10ff231518
# Parent  583830ce6afaea73a944242d4113dc85a373fc63
onenormest.m: Initialize random number generator in tests to guarantee reliable passage.

* onenormest.m: Initialize random number generator in tests to guarantee reliable passage.

diff --git a/scripts/linear-algebra/onenormest.m b/scripts/linear-algebra/onenormest.m
--- a/scripts/linear-algebra/onenormest.m
+++ b/scripts/linear-algebra/onenormest.m
@@ -273,15 +273,16 @@ endfunction
 %!  assert (nm1, norm (A, 1), -2*eps);
 %!  assert (nminf, norm (A, inf), -2*eps);
 %!  assert (norm (w1, 1), nm1 * norm (v1, 1), -2*eps)
 %!  assert (norm (winf, 1), nminf * norm (vinf, 1), -2*eps)
 
 ## Only likely to be within a factor of 10.
 %!test
 %!  N = 100;
+%!  rand ('state', 42);  % Initialize to guarantee reproducible results
 %!  A = rand (N);
 %!  [nm1, v1, w1] = onenormest (A);
 %!  [nminf, vinf, winf] = onenormest (A', 6);
 %!  assert (nm1, norm (A, 1), -.1);
 %!  assert (nminf, norm (A, inf), -.1);
 %!  assert (norm (w1, 1), nm1 * norm (v1, 1), -2*eps)
 %!  assert (norm (winf, 1), nminf * norm (vinf, 1), -2*eps)
